.github/workflows/lint.yml

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+name: Lint all codez
+on:
+  push:
+    tags: 'release-*'
+    branches:
+      - dev
+      - release
+  pull_request:
+    branches:
+      - main
+      - dev
+
+jobs:
+  go:
+    name: Lint go files
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.13.6
+      id: go
+
+    - name: install crlfmt
+      run: |
+        # Workaround to download the specific version we use internally
+        cat > go.mod << EOF
+        module vectorized
+        require (
+          github.com/cockroachdb/crlfmt v0.0.0-20200309141329-97b747f03b21
+        )
+        go 1.13
+
+        EOF
+
+        go mod download
+        go install github.com/cockroachdb/crlfmt
+        rm -rf go.mod
+
+    - name: lint go files
+      run: |
+        find . -name *.go -type f | xargs -n1 crlfmt -wrap=80 -w
+        git diff --exit-code
+
+  js:
+    name: Lint js files
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Set up Node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: '12.16.1'
+
+    - name: Set up prettier
+      run: npm install prettier
+
+    - name: Lint JS files
+      run: npx prettier --list-different src/**/*.{ts,js}
+
+  py:
+    name: Lint python files
+    runs-on: ubuntu-20.04
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Set up python
+      run: |
+        sudo apt-get install -y python3 python3-pip
+        sudo pip install yapf
+
+    - name: lint python files
+      run: find . -name *.py -type f | xargs -n1 yapf -d
+
+  cpp:
+    name: Lint files with clang-format
+    runs-on: ubuntu-latest
+    # Use purpose built container since the latest llvm versions we use are
+    # not always available as packages. This container can also get used by contributors locally.
+    container: 0xdiba/clang-format:latest
```

Comment:
i wonder if llvm maintains this. we use clang-format-11, we can use upstream or chrome nightly builds on clang-format-11 tbh. we don't have to maintain our own docker img.

Reply:
We build it - it should be in gcr.io/redpandaci/clang

Reply:
```
docker run -it gcr.io/redpandaci/clang:d798e62dfa /vectorized/llvm/bin/clang-format --version
clang-format version 11.0.0 (https://2b06bc13d5b28b10f180d82c9f4096e641a4cc9f@github.com/0xdiba/meta 9f689577385056a1b26338f423976585788d0c65)
```

Reply:
Oh, you want something on [hub.docker.com](hub.docker.com)

Reply:
good point, i thought gcr is not avail outside of CI ?

Reply:
```
 docker run -it gcr.io/redpandaci/clang:d798e62dfa /vectorized/llvm/bin/clang-format --version
Unable to find image 'gcr.io/redpandaci/clang:d798e62dfa' locally
docker: Error response from daemon: unauthorized: You don't have the needed permissions to perform this operation, and you may have invalid credentials. To authenticate your request, follow the steps in: https://cloud.google.com/container-registry/docs/advanced-authentication.
See 'docker run --help'.```

Reply:
we want that. , so yeah, we can publish the binary to docker hub for public access.

Reply:
Had checked chromium ,as I saw they just [build against a specific ref](https://chromium.googlesource.com/chromium/src/+/master/buildtools/DEPS#20)
Tested with the prebuilt release at https://releases.llvm.org/download.html, had some problems and after tinkering a bit with it the [clang-format step took 2,5 minutes ](https://github.com/0xdiba/redpanda/runs/1446704677?check_suite_focus=true)(between updating packages, downloading some missing libs and fetching the prebuilt binaries) which is a lot compared to the 19s it takes with the bare bones container.

The GCR container we build is a great idea but imo has a couple of dealbreakers:
* Contributors cannot take the command and use it locally as is
* We would need to import credentials in GitHub or opensource it.
* Since it fetches and builds the whole clang codebase, it is a bunch of GBs which is probably a waste.

Reply:
seems like you could build an ubuntu:groovy docker image on-demand with clang-format (clang 11 appears to be available) and then either do that for each run or stash the image in the github container registry that's available. another option would be to download the pre-built clang tarball and stash away the clang-format binary using github cache action.

Reply:
great idea on the github cache action.

Reply:
Went with Noah's suggestion and made the linting working on ubuntu:groovy. Time is not that much of an issue with that setup so I think our issue is solved.